c7n/resources/iam.py

```diff
@@ -817,6 +817,41 @@ def process(self, resources):
                     pass
 
 
+@Role.action_registry.register('delete')
+class RoleDelete(BaseAction):
+    """Delete an IAM Role.
+
+    For example, if you want to automatically delete an unused IAM role.
+
+    :example:
+
+      .. code-block:: yaml
+
+        - name: iam-delete-unused-role
+          resource: iam-role
+          filters:
+            - type: unused
```

Comment:
please switch the example to using usage, ... unused isn't really a safe filter, for many usages as it only considers ec2, ecs, lambda usages, and there are many other usage modes. the usage filter will look at actual api usage of the role.

Reply:
Switched to type:usage